File: 25.合并两个排序的链表/25.合并两个排序的链表.py
# -*- coding:utf-8 -*-
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
class Solution:
    # 返回合并后列表
    def Merge(self, pHead1, pHead2):
        # write code here
        if pHead1 ==None and pHead2 ==None:
            return None
        if pHead1 == None:
            return pHead2
        if pHead2 == None:
            return pHead1
        p = pHead2.next
        q = pHead1
        front = pHead2
        if(pHead1.val<pHead2.val):
            front = pHead1
            p = pHead1.next
            q = pHead2
        head = front
        while(p!=None and q!=None):
            if(q.val<p.val):
                temp = q.next
                q.next = p
                front.next = q
                front = q
                q=temp
            else:
                front = front.next
                p = p.next
        if(q!=None):
			front.next = q            
                
        return head

File: 25.合并两个排序的链表/25.合并两个排序的链表.cpp
/*
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};*/
class Solution {
public:
    ListNode* Merge(ListNode* pHead1, ListNode* pHead2)
    {
        ListNode* head = new ListNode(-1);
        head->next = nullptr;
        ListNode* cur = head;
        while(pHead1 != nullptr and pHead2!=nullptr){
            if (pHead1->val < pHead2->val){
                ListNode* temp = pHead1->next;
                cur->next = pHead1;
                pHead1 = temp;
            }else{
                ListNode* temp = pHead2->next;
                cur->next = pHead2;
                pHead2 = temp;
            }
            cur = cur->next;
        }
        cur->next = nullptr;
        if (pHead1 != nullptr)
            cur->next = pHead1;
        if (pHead2 != nullptr)
            cur->next = pHead2;
        return head->next;
    }
};